yb-voyager/src/cp/yugabyted/yugabyted.go

```diff
@@ -0,0 +1,678 @@
+/*
+Copyright (c) YugabyteDB, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package yugabyted
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	log "github.com/sirupsen/logrus"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/cp"
+	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
+)
+
+type YugabyteD struct {
+	sync.Mutex
+	conn               *pgxpool.Pool
+	migrationDirectory string
+}
+
+func New(exportDir string) *YugabyteD {
+	return &YugabyteD{migrationDirectory: exportDir}
+}
+
+// Initialize the target DB for visualisation metadata
+func (cp *YugabyteD) Init() error {
+	err := cp.connect()
+	if err != nil {
+		return err
+	}
```

Comment:
Add context message to the `err` before returning it. Refer uses of `fmt.Errorf()` in other voyager code. Fix all such occurrences.